File: Yandex/Practicum/Algorightms_and_Data_Structures/3_sprint/D.Печеньки/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>


void solution(int& n, std::vector<int>& greedy, int& m, std::vector<int>& cookies) {
    /*
    Time Complexity O(NlogN)
    Memory Complexity O(1)
    */
    int happy = 0;
    std::sort(greedy.begin(), greedy.end());
    std::sort(cookies.begin(), cookies.end());
    int cookiesIdx = 0;
    for (int gred : greedy) {
        while (cookiesIdx < m && cookies[cookiesIdx] < gred) {
            ++cookiesIdx;
        }
        if (cookiesIdx < m) {
            ++happy;
            ++cookiesIdx;
        } else {
            break;
        }
    }
    std::cout << happy << "\n";
}


int main() {
    int n;
    std::cin >> n;
    std::vector<int> greedy;
    for (int i = 0; i < n; ++i) {
        int gred;
        std::cin >> gred;
        greedy.push_back(gred);
    }
    int m;
    std::cin >> m;
    std::vector<int> cookies;
    for (int i = 0; i < m; ++i) {
        int cookie;
        std::cin >> cookie;
        cookies.push_back(cookie);
    }
    solution(n, greedy, m, cookies);
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/3_sprint/D.Печеньки/solution.py
import sys


def solution(n: int, greedy: list[int], m: int, cookies: list[int]) -> None:
    """
    Time Complexity O(NlogN)
    Memory Complexity O(1)
    """
    happy: int = 0
    greedy.sort()
    cookies.sort()
    cookies_idx: int = 0
    for gred in greedy:
        while cookies_idx < m and cookies[cookies_idx] < gred:
            cookies_idx += 1
        if cookies_idx < m:
            happy += 1
            cookies_idx += 1
        else:
            break
    sys.stdout.write(str(happy) + '\n')


if __name__ == '__main__':
    n: int = int(sys.stdin.readline().rstrip())
    greedy: list[int] = list(map(int, sys.stdin.readline().rstrip().split()))
    m: int = int(sys.stdin.readline().rstrip())
    cookies: list[int] = list(map(int, sys.stdin.readline().rstrip().split()))
    solution(n, greedy, m, cookies)
